observability/config.go

```diff
@@ -0,0 +1,109 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package observability
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	gcplogging "cloud.google.com/go/logging"
+	"golang.org/x/oauth2/google"
+	configpb "google.golang.org/grpc/observability/internal/config"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+const (
+	envKeyObservabilityConfig = "GRPC_OBSERVABILITY_CONFIG"
+)
+
+// gcpDefaultCredentials is the JSON loading struct used to get project id.
+type gcpDefaultCredentials struct {
+	QuotaProjectID string `json:"quota_project_id"`
+}
+
+// fetchDefaultProjectID fetches the default GCP project id from environment.
+func fetchDefaultProjectID(ctx context.Context) string {
+	// Step 1: Check ENV var
+	if s := os.Getenv("GCLOUD_PROJECT_ID"); s != "" {
+		return s
+	}
+	// Step 2: Check default credential
+	if credentials, err := google.FindDefaultCredentials(ctx, gcplogging.WriteScope); err == nil {
+		logger.Infof("found Google Default Credential")
```

Comment:
Throughout: log messages should always start with a capital letter (unless it's a word that must not be capitalized like an unexported function name or "gRPC").  Error messages should begin lowercase (with the same exceptions), because they may be concatenated.

Reply:
> By log messages, do you mean messages printed by logger, regardless it's a debug log or error log

Yes.

> And by error messages, you mean only fmt.Errorf that will be returned between functions?

There are other ways to produce `error`s (e.g. `errors.New` or `status.Error[f]`), but yes.

> Double-checking: I was following the pattern used by "core" logger. It seems to use lowercase in any case?

Well that's unfortunate.  Don't blindly copy code you see elsewhere, I guess.  😄